src/car.cpp

```diff
@@ -816,8 +816,12 @@ bool Car::check_mesh_collision(vec2 test_vertex, Triangle t) {
 	float collisionArea2 = get_triangle_area(t.a, test_vertex, t.c);
 	float collisionArea3 = get_triangle_area(test_vertex, t.b, t.c);
 	float collisionSum = collisionArea1 + collisionArea2 + collisionArea3;
+	if (tArea == 0.f) {
+		printf("why Triangle: (%f %f) (%f %f) (%f %f) Vertex: (%f, %f)\n", t.a.x, t.a.y, t.b.x, t.b.y, t.c.x, t.c.y, test_vertex.x, test_vertex.y); 
+	}
 	//printf("Second car tArea: %f   collisionArea: %f \n", tArea, collisionSum);
-	return (abs(collisionSum - tArea) <= 0.1f);
+
+	return (abs(collisionSum - tArea) <= 2.0f);

```

Comment:
did this make a difference for you? 0.1f was working fairly well for me but that's before you caught the world coordinate bug

Reply:
Eh, I figured it doesn't hurt to be no the safe side but it probably makes no difference. I'll lower it to 1.0f, just in case of really bad float precision.